scipy/stats/morestats.py

```diff
@@ -1027,14 +1031,22 @@ def boxcox(x, lmbda=None, alpha=None):
     if x.size == 0:
         return x
 
-    if any(x <= 0):
+    contains_nan, nan_policy = _contains_nan(x, nan_policy)
+    not_nan_mask = ~np.isnan(x)
+
+    if any(x[not_nan_mask] <= 0):
         raise ValueError("Data must be positive.")
 
     if lmbda is not None:  # single transformation
-        return special.boxcox(x, lmbda)
+        return special.boxcox(x, lmbda)  # propagates nan's
```

Comment:
I think it should be

```python
if nan_policy == 'propagate':
    return special.boxcox(x, lmbda)
else:
    return special.boxcox(x[not_nan_mask], lmbda)
```

It won't broadcast `lmbda` correctly, but that doesn't seem like an intended use case. @josef-pkt or @ev-br would know better than me though, so we should see what they say.

Reply:
Great finding, thank you.
To achieve the same, I just moved code for omitting nan values.
Check please, if it meets your expectations.